### ISSUE TITLE
Fatal crash when sending a message

### ISSUE THREAD

**Issue by jlara310** [GoogleCloudPlatform/firebase-android-client#13]
App crashes in emulator when sending a message:

`E/AndroidRuntime: FATAL EXCEPTION: main
                  Process: com.google.cloud.solutions.flexenv, PID: 6144
                  java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.google.android.gms.auth.api.signin.GoogleSignInAccount.getDisplayName()' on a null object reference
                      at com.google.cloud.solutions.flexenv.PlayActivity.onKey(PlayActivity.java:268)`

I tried replacing the call to getDisplayName() with a constant and that removes the error.

**Comment by ricalo** [comment]
Nevermind, I was able to reproduce the error.
It  looks like the `acct` object is not initialized unless the user explicitly goes through the sign in flow. However, Firebase remembers users already signed in, returning users don't need to go through the sign in flow.

The activity should get the user information for signed in users.

Additionally, the `acct` object is of type [`GoogleSignInAccount`](https://developers.google.com/android/reference/com/google/android/gms/auth/api/signin/GoogleSignInAccount), but it would be better to use a [`FirebaseUser`](https://firebase.google.com/docs/reference/android/com/google/firebase/auth/FirebaseUser) where possible to make the sample more flexible.

Working on this, let me know if you have any comments.

**Comment by jlara310** [comment]
Thanks Ricardo! I confirmed that the updated code allows me to send and view messages without crashing the app.

I noticed another small bug when you sign out and back in, not sure if it's related:

1. Click Sign Out
2. Click Sign in

At this point the app hangs while displaying the message, "Authenticating with Firebase..." If I close and re-open the app, though, I see that I'm succesfully signed in and able to send messages.

Should I opened a separate issue for this?
